### PR TITLE
8 cloning an existing repository

### PR DESCRIPTION
**Pull request by Lsnaathorst1**
Adding cloning an existing repository information 

(May want to change which headers automatically get a line/don't get a line in future!)

## Overview of changes
Guide and tasks for cloning an existing repo 

## Issue ticket number/s and link
Cloning an existing repository #8

### CODE REVIEW

File: ch2-git-basics/cloning.qmd [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+There are several ways to clone an existing repository. There are two methods detailed below. The first uses Rstudio and the second uses Git Bash. Start with step 1, which is the same for both, and then decide which method to follow. Once you have successfully cloned, you should see the files appear in both the file you have chosen to store your repository in and the files pane in R studio.
+
+### Step 1: Clone the Repository
+
+a\. Go to the [git-academy-sandbox](https://github.com/dfe-analytical-services/git-academy-sandbox) repository page on GitHub.
+
+b\. Click the "Code" green button and copy the URL (HTTPS link).
+
+### To Clone Using RStudio:
```
Comment (rmbielby) [inline ch2-git-basics/cloning.qmd:15]: How about we try using a tabset panel for the different methods?

https://quarto.org/docs/interactive/layout.html#tabset-panel

File: ch2-git-basics/cloning.qmd [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+**git clone**
+
+**https://github.com/dfe-analytical-services/git-academy-sandbox.git**
+
+You will need to right-click to paste, as the keyboard shortcut will not work in Git Bash.
+
+#### Step 2c: Verify the Clone
+
+Check the Files pane in RStudio to see if the repository files have been cloned to your local machine. The files pane normally opens in the right-hand bottom corner of RStudio.
```
Comment (rmbielby) [inline ch2-git-basics/cloning.qmd:53]: I think for this one we should be IDE agnostic. Analysts could be working in R, Python, SQL, html+css, etc, so for the git bash one, I think we shouldn't use R-Studio at all.

Instructions for checking the repo in git bash would be along the lines of:

- Navigate into the repo directory, e.g. `cd git-academy-sandbox`
- You should now see the current branch name (e.g. main) at the end of the filepath
- type `git log` to view recent commits or `git status` to check the state of the cloned repository (at this point it should say everything's up to date as you haven't made any changes to anything yet)

File: ch2-git-basics/cloning.qmd [hunk 3]
```diff
@@ -0,0 +1,57 @@
+## Cloning an Existing Repository
+
+### Introduction
+
+This section will guide you through the process of cloning a GitHub repository. Cloning a repository means making a local copy of a project that exists on GitHub. This enables you to make changes and test locally before pushing to be approved and merged with the main branch on GitHub.
```
Comment (rmbielby) [inline ch2-git-basics/cloning.qmd:5]: "that exists on GitHub" -> "that exists on a "remote" DevOps server such as Axure DevOps, GitHub or GitLab"

**Review by rmbielby — CHANGES_REQUESTED** [review]
Looks good. Few minor comments / suggestions

**Comment by Lsnaathorst1** [comment]
@rmbielby suggested changes made and tried to tidy the headings a bit too :)

File: ch2-git-basics/cloning.qmd [hunk 4]
```diff
@@ -0,0 +1,69 @@
+---
+title: "Cloning an Existing Repository"
+---
+
+## Introduction
+
+This section will guide you through the process of cloning a GitHub repository. Cloning a repository means making a local copy of a project that exists on a "remote" DevOps server such as Axure DevOps, GitHub or GitLab.
```
Comment (rmbielby) [inline ch2-git-basics/cloning.qmd:7]: Typo: Axure -> Azure

File: ch2-git-basics/cloning.qmd [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+
+This enables you to make changes and test locally before pushing to be approved and merged with the main branch on GitHub.
+
+There are several ways to clone an existing repository. Below, you'll find two methods. The first uses Rstudio and the second uses Git Bash. Start with step 1, which is the same for both methods, and then decide which method to follow. Once you have successfully cloned, you should see the files appear in the location you have chosen to store your repository.
+
+## Step 1: Clone the Repository
```
Comment (rmbielby) [inline ch2-git-basics/cloning.qmd:13]: This feels more like "Get the repository clone link" rather then "Clone the repository"

File: ch2-git-basics/cloning.qmd [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+
+a\. Go to the [git-academy-sandbox](https://github.com/dfe-analytical-services/git-academy-sandbox) repository page on GitHub.
+
+b\. Click the "Code" green button and copy the URL (HTTPS link).
+
+::: panel-tabset
+## To Clone Using RStudio:
+
+### Step 2a: Open RStudio
```
Comment (rmbielby) [inline ch2-git-basics/cloning.qmd:22]: I don't want to be R centric, so can we shift Git bash to be the first panel.

File: ch2-git-basics/cloning.qmd [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+https://github.com/dfe-analytical-services/git-academy-sandbox.git
+```
+
+You will need to right-click to paste, as the keyboard shortcut will not work in Git Bash.
+
+### Step 2c: Verify the clone in Git Bash
+
+-   Navigate into the repo by typing **cd git-academy-sandbox**.
+
+-   You should now see the current branch name (e.g. main) at the end of the file path.
```
Comment (rmbielby) [inline ch2-git-basics/cloning.qmd:62]: Could add a small image to show what this looks like.

**Review by rmbielby — CHANGES_REQUESTED** [review]
Nice. Just a few more minor comments though! :-)

File: ch2-git-basics/cloning.qmd [hunk 8]
```diff
@@ -0,0 +1,69 @@
+---
+title: "Cloning an Existing Repository"
+---
+
+## Introduction
+
+This section will guide you through the process of cloning a GitHub repository. Cloning a repository means making a local copy of a project that exists on a "remote" DevOps server such as Axure DevOps, GitHub or GitLab.
+
+This enables you to make changes and test locally before pushing to be approved and merged with the main branch on GitHub.
```
Comment (rmbielby) [inline ch2-git-basics/cloning.qmd:9]: Wonder if this is too much extra info. They may not know about pushing, merging and approving yet. I'd be tempted to bring it back to the very specific basics:

"If you have a repository on Azure DevOps, GitHub or GitLab, you generally can't edit or run your work on those platforms, so if you need to work with the contents of a repo, you'll usually need to make a clone."

File: ch2-git-basics/cloning.qmd [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+
+There are several ways to clone an existing repository. Below, you'll find two methods. The first uses Rstudio and the second uses Git Bash. Start with step 1, which is the same for both methods, and then decide which method to follow. Once you have successfully cloned, you should see the files appear in the location you have chosen to store your repository.
+
+## Step 1: Clone the Repository
+
+a\. Go to the [git-academy-sandbox](https://github.com/dfe-analytical-services/git-academy-sandbox) repository page on GitHub.
+
+b\. Click the "Code" green button and copy the URL (HTTPS link).
```
Comment (rmbielby) [inline ch2-git-basics/cloning.qmd:17]: The colour of the button depends on your theme (mine's blue), so maybe just remove green here.

File: ch2-git-basics/cloning.qmd [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+
+Navigate to the folder where you want to clone the repository. Right-click in the directory and select "Git Bash Here".
+
+### Step 2b: Clone the Repository in Git Bash
+
+-   In the Git Bash window that opens, use the \`git clone\` command followed by the URL you copied from GitHub. For this repository, we would write:
+
+``` bash
+git clone
+
```
Comment (rmbielby) [inline ch2-git-basics/cloning.qmd:52]: This needs to be all one line

File: ch2-git-basics/cloning.qmd [hunk 11]
```diff
@@ -0,0 +1,69 @@
...
+You will need to right-click to paste, as the keyboard shortcut will not work in Git Bash.
+
+### Step 2c: Verify the clone in Git Bash
+
+-   Navigate into the repo by typing **cd git-academy-sandbox**.
+
+-   You should now see the current branch name (e.g. main) at the end of the file path.
+
+-   Type **git log** to view recent commits or **git status** to check the state of the cloned repository (at this point, it should say everything is up to date, as you haven't made any changes yet).
```
Comment (rmbielby) [inline ch2-git-basics/cloning.qmd:64]: Need to add "(and press q to exit git log)" after "to view recent commits"

**Comment by Lsnaathorst1** [comment]
@rmbielby I think this is ready for another review when you're ready

**Review by rmbielby — APPROVED** [review]

**Pull request merged**